src/components/Garden/ModalFlows/CreateProposalScreens/CreateProposalScreens.tsx

```diff
@@ -93,32 +93,35 @@ function CreateProposalScreens({ onComplete }) {
     async (onComplete) => {
       const { amount, beneficiary, link, title } = proposalData.current
 
-      let params
-      let fn
       if (amount.valueBN.eq(0)) {
-        fn = 'newSignalingProposal'
-        params = {
-          title,
-          link,
-        }
+        await convictionActions.newSignalingProposal(
+          {
+            title,
+            link,
+          },
+          (intent: any) => {
+            setTransactions(intent)
+            onComplete()
+          }
+        )
       } else {
         const convertedAmount = amount.valueBN.toString(10)
         const stableRequestAmount = amount.stable
 
-        fn = 'newProposal'
-        params = {
-          title,
-          link,
-          amount: convertedAmount,
-          stableRequestAmount,
-          beneficiary,
-        }
+        await convictionActions.newProposal(
+          {
+            title,
+            link,
+            amount: convertedAmount,
+            stableRequestAmount,
+            beneficiary,
+          },
+          (intent: any) => {
```

Comment:
Looks great and more safe that way, what i'll suggest is encapsulate the callback

```
const callbackSomethingNameHere = (intent: any) => {
  setTransactions(intent)
            onComplete()
          }
```

So it will keep equal the previous one and avoid update one and forget the other.

Reply:
Yep makes sense. will update that.

Reply:
Just updated!